app/controllers/subscriptions_management_controller.rb

```diff
@@ -1,6 +1,7 @@
 class SubscriptionsManagementController < ApplicationController
   include Slimmer::Headers
   include Slimmer::Template
+  include GovukPersonalisation::ControllerConcern
```

Comment:
We don't need to pull in the personalisation controller concern here, as email-alert-frontend has its own session cookie - rather than replace that, we're just adding a flag to it when the user is logged in via an account.

Reply:
Ah ok. I thought we needed to 'import' the module in some way for the controller to be able to see it, but it looks like Ruby is smarter than Python in that regard

Reply:
Rails is importing the gem automatically, and in the top-level module of the gem [we import all the modules](https://github.com/alphagov/govuk_personalisation/blob/main/lib/govuk_personalisation.rb).  An `import` in Ruby is less like a Python-style `import` and more like a C-style `#include` (kind of).

`include` merges a module into a different class / module, making all the things defined in the `include`d module available in the `include`ing one.

Reply:
It's mostly "Rails magic" I think.
Rails monitors certain folders and filenames and autoloads them using "magic" (... which i think means metaprogramming)